examples/SLAM/js/spawn-mesh-on-reticle.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, Material, Mesh, Type } from '@wonderlandengine/api';
+import { ARSession, ARProvider, WebXRProvider } from '../../../';
+class SpawnMeshOnReticle extends Component {
+
+  public static TypeName = 'spawn-mesh-on-reticle';
+  public static Properties = {
+    /* The mesh to spawn */
+    mesh: { type: Type.Mesh },
+    /* The material to spawn the mesh with */
+    material: { type: Type.Material },
+  };
+
+  mesh!: Mesh;
+  material!: Material;
+
+  start() {
+    ARSession.onSessionStarted.push(this.onSessionStarted);
+    ARSession.onSessionEnded.push(this.onSessionEnded);
+  }
+
+  onSessionStarted = (provider: ARProvider) => {
+    /* We set this function up to get called when a session starts.
+     * The 'select' event happens either on touch or when the trigger
+     * button of a controller is pressed.
+     * Once that event is triggered, we want spawnMesh() to be called. */
+    if (provider instanceof WebXRProvider) {
+      (provider as WebXRProvider).xrSession!.addEventListener('select', this.spawnMesh);
+    } else {
+      window.addEventListener('click', this.spawnMesh);
+    }
+  }
+
+  onSessionEnded = (provider: ARProvider) => {
+    if (provider instanceof WebXRProvider) {
+      (provider as WebXRProvider).xrSession!.removeEventListener('select', this.spawnMesh);
```

Comment:
> Let session be a [new](https://webidl.spec.whatwg.org/#new) [XRSession](https://immersive-web.github.io/webxr/#xrsession) object in the [relevant realm]

From [the WebXR Device API spec](https://immersive-web.github.io/webxr/#dom-xrsystem-requestsession). But nothing on `.end()` indicates that listeners would be removed.

It's a bit strange, while this means that the callbacks are not reused for the next session, the ended session is kept alive and will retain references to the listeners (which in return might retain references to large data etc). Maybe add a comment here that removing the event listener is indeed necessary for being able to garbage collect any unused memory referenced by the event listener.